tests/pass-dep/libc/pthread-threadname.rs

```diff
@@ -199,4 +199,28 @@ fn main() {
         .unwrap()
         .join()
         .unwrap();
+
+    // Now set the name for a non-existing thread and verify error codes.
+    cfg_if::cfg_if! {
+        if #[cfg(any(target_os = "linux", target_os = "illumos", target_os = "solaris", target_os = "macos"))] {
```

Comment:
This has no good reason to be `cfg_if` either, you can make it a block.

Also, can you make this `cfg(not(any(...)))` to ensure that when we add new OSes, we consider the question whether they need to be checked here?